### PR TITLE
chore: update changelog and versioning for 1.1.12 release

### PR DESCRIPTION
**Pull request by aaronchung-bitquill**
### Summary

chore: update changelog and versioning for 1.1.12 release

### Description

chore: update changelog and versioning for 1.1.12 release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
 
+## [1.1.12] - 2023-12-21
+### Added
+* Performance test workflow ([PR #517](https://github.com/awslabs/aws-mysql-jdbc/pull/517)).
+* [**Experimental** Enhanced Host Monitoring Plugin v2](https://github.com/awslabs/aws-mysql-jdbc/blob/main/README.md#experimental-enhanced-failure-monitoring-plugin-v2) , which is a redesign of the original Enhanced Host Monitoring Plugin that addresses memory leaks and high CPU usage during monitoring sessions ([PR #513](https://github.com/awslabs/aws-mysql-jdbc/pull/513)).
```
Comment (sergiyvamz) [inline CHANGELOG.md:9]: Enhanced Host Monitoring Plugin that addresses ->
Enhanced Host Monitoring Plugin and addresses
?

File: CHANGELOG.md [hunk 2]
```diff
@@ -183,6 +188,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Added
 * This driver is based on the MySQL 8.0.21 community driver. The driver is cluster aware for Amazon Aurora MySQL. It takes advantage of Amazon Aurora's fast failover capabilities, reducing failover times from minutes to seconds.
 
+[1.1.11]: https://github.com/awslabs/aws-mysql-jdbc/compare/1.1.11...1.1.12
```
Comment (sergiyvamz) [inline CHANGELOG.md:191]: 1.11.12

File: CHANGELOG.md (hunk 1 above)
Comment (karenc-bq) [reply CHANGELOG.md:9]: Not sure if we need to add performance test workflow in the changelog

Comment (aaronchung-bitquill) [reply CHANGELOG.md]: Done!

File: CHANGELOG.md (hunk 2 above)
Comment (aaronchung-bitquill) [reply CHANGELOG.md]: Fixed!

**Review by sergiyvamz — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (aaronchung-bitquill) [reply CHANGELOG.md]: Didn't see your message @karenc-bq . 
I've now removed performance test workflow.

**Pull request merged**